python-service/src/health_aggregator.py

```
@@ -0,0 +1,97 @@
+            f"{self.coordinator_url}/nodes/{node_id}/health",
+            method="GET",
+        )
+        try:
+            response = urllib.request.urlopen(req, timeout=3)
+            return json.loads(response.read().decode("utf-8"))
+        except urllib.error.URLError:
+            return {}
+
+    def fetch_aggregate_metrics(self, service_name: str, region: str) -> Dict[str, Any]:
+        url = f"{self.coordinator_url}/metrics/{service_name}?region={region}"
+        req = urllib.request.Request(url, method="GET")
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)